PYTHON_3_13_UPGRADE_SUMMARY.md

```diff
@@ -0,0 +1,138 @@
+
+## Overview
+
+This document summarizes all changes made to update the sadie-antibody project for Python 3.13 compatibility. The project has been successfully updated from supporting Python 3.9-3.10 to Python 3.9-3.13.
+
+## Changes Made
+
+### 1. Python Version Support Updates
+
+#### pyproject.toml
+- Updated Python classifiers to include Python 3.11, 3.12, and 3.13
+- Removed deprecated Python 3.8 classifier
+- Python constraint already supported: `>=3.9,<=3.13` ✅
+
+#### GitHub Actions Workflows
+- All workflows were already testing against Python 3.13 ✅
+- Test matrix includes: [3.9, '3.10', '3.11', '3.12', '3.13']
+- Poetry version updated to 1.8
+
+#### Environment Configuration
+- environment.yml already specified `python>=3.13` ✅
+- All dependencies already compatible with Python 3.13 ✅
+
+### 2. Pydantic v1 to v2 Migration
+
+The project was using Pydantic v2 constraint (`>=2.0.0,<3.0.0`) but some code was still using v1 patterns.
+
+#### Core Model Files Updated
+- `src/sadie/reference/models.py`:
+  - Replaced `@validator` with `@field_validator`
+  - Updated method signatures to use `@classmethod`
+  - Removed `values` parameter access
+
+- `src/sadie/airr/models/series.py`:
+  - Replaced `@root_validator` with `@model_validator`
+  - Updated to use `mode='before'` parameter
+
+#### Custom Type Validators Updated
+- `src/sadie/typing/species.py`:
+  - Replaced `ModelField` and `__get_validators__` with `__get_pydantic_core_schema__`
+  - Updated to use `pydantic_core.core_schema`
+
+- `src/sadie/typing/chain.py`:
+  - Same Pydantic v2 migration pattern
+
+- `src/sadie/typing/source.py`:
+  - Same Pydantic v2 migration pattern
+
+#### Deprecated Decorator Removal
+- Removed `@validate_arguments` decorator from:
+  - `src/sadie/renumbering/aligners/hmmer.py`
+  - `src/sadie/renumbering/clients/g3.py`
+  - `tests/unit/typing/test_typing.py`
+
+- Updated test patterns to use Pydantic models instead of function decorators
+
+### 3. Version Detection (Already Compatible)
+
+The main application (`src/sadie/app.py`) already uses the modern approach:
+```python
+try:
+    from importlib.metadata import version
+except ImportError:
+    # Fallback for Python < 3.8
+    from importlib_metadata import version
+```
+
+This correctly uses `importlib.metadata` instead of the deprecated `pkg_resources`.
+
+### 4. Documentation Updates
+
+#### README and Documentation
+- Updated Python version badges to show 3.9-3.13 support
+- Updated installation instructions
+- Updated contribution guide
+
+#### Development Setup
+- `src/sadie/typing/README.md`: Updated example code for Pydantic v2
+
+### 5. Legacy Code (Not Updated)
+
+Files in `antibody_objects_old/` directory contain legacy code using deprecated `pkg_resources`, but these appear to be old/unused code that doesn't affect the main application.
+
+## Compatibility Verification
+
+### Dependencies Status
+All major dependencies are compatible with Python 3.13:
+- ✅ biopython: ^1.84
```

Comment:
@kipkurui biopython 1.84 is the cut-off for import migrations. so far this is the new 1.80 for awhile hopefully